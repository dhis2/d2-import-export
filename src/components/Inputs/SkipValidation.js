import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_YES = { value: 'true', label: i18n.t('Yes') }
export const OPTION_NO = { value: 'false', label: i18n.t('No') }
export const SKIP_VALIDATION_KEY = 'skipValidation'
export const SKIP_VALIDATION_DEFAULT_VALUE = OPTION_NO.value

export const SkipValidation = () => (
    <RadioGroup
        name={SKIP_VALIDATION_KEY}
        label={i18n.t('Skip validation')}
        options={[OPTION_YES, OPTION_NO]}
    />
)
