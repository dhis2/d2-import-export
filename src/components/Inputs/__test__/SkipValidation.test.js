import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { SkipValidation } from '../SkipValidation'

jest.mock('react-final-form')

describe('Input component - SkipValidation', () => {
    it('should render correctly', () => {
        const file = shallow(<SkipValidation />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
