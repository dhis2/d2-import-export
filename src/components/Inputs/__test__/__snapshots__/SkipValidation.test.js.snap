// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - SkipValidation should render correctly 1`] = `
<Field>
  <Label>
    Skip validation
  </Label>
  <RadioGroup
    dataTest="input-skip-validation"
    label="Skip validation"
    name="skipValidation"
    options={
      Array [
        Object {
          "label": "Yes",
          "value": "true",
        },
        Object {
          "label": "No",
          "value": "false",
        },
      ]
    }
  />
</Field>
`;
